### PR TITLE
Add script to cleanup github teams.

### PR DESCRIPTION
**Pull request by feanil**
Output from the dry-run: [team_dry_run.txt](https://github.com/openedx/terraform-github/files/8003784/team_dry_run.txt)

### CODE REVIEW

**Review by Carlos-Muniz — APPROVED** [review]
The script functions look good but are starting to get a bit bulky. At what point do we add linting and other helper packages?

File: migrate/remove_unused_teams.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                api.teams.delete_in_org(org, team)
+            click.secho(f"Done", fg="red")
+            deletion_count += 1
+
+    for team in members_but_nothing_else:
+        click.secho(f"'{team}' has the following members:", bold=True)
+        click.secho("    - ", nl=False)
+        click.secho("\n    - ".join(team_to_members[team]))
+
+        if dry_run or click.confirm(f"Delete '{team}'?"):
```
Comment (ormsbee) [inline migrate/remove_unused_teams.py:104]: It's okay to run through these individually? I feel like this is the sort of thing I'd be afraid of messing up if I was typing it live with a non-trivial number of teams.

**Review by ormsbee — CHANGES_REQUESTED** [review]
I have a number of comments, but they're basically all optional and dependent on how long we want to keep this code around. One question I did have though: Do we have a good, current backup of the team information in case we wipe out something we didn't mean to during this process and need to restore some (even if that restoration is manual)? (I guess in the worst case, we could go to the edX org?)

File: migrate/remove_unused_teams.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        child_teams = api.teams.list_child_in_org(org, team.slug)
+
+        # We get all the team members so we can review teams that still
+        # have members.
+        team_members = []
+        for page in paged(api.teams.list_members_in_org, org, team.slug):
+            team_members.extend(page)
+
+        team_to_members[team.slug] = [member.login for member in team_members]
+        if len(repos) == 0 and len(projects) == 0 and len(child_teams) == 0:
```
[the remark this answers is not in the document]
Comment (ormsbee) [reply migrate/remove_unused_teams.py:43]: Or I guess:

```python
if not (has_repos or has_projects or has_child_teams):
```

File: migrate/remove_unused_teams.py (hunk 1 above)
Comment (feanil) [reply migrate/remove_unused_teams.py:104]: yea, I was debating this part, an alternative I considered is that I could turn this part into a separate script that reports the data and one that will delete teams on a given list.  I think it's not too scary but I'm happy to reconsider it.

**Review by feanil — COMMENTED** [review]
Thanks for the reviews, I'm making most of the suggested changes, but wanted to talk more about the cleanup of teams that have members but no associated repos.

**Review by ormsbee — APPROVED** [review]

File: migrate/remove_unused_teams.py (hunk 1 above)
Comment (ormsbee) [reply migrate/remove_unused_teams.py]: Given that the failure scenario isn't hard to recover from, I agree–not too scary.

**Pull request merged**